server/api/translations.js

```diff
@@ -90,11 +89,61 @@ router.put(
   }
 );
 
+router.delete(
+  `/:id`,
+  passport.authenticate("jwt", { session: false }),
+  (req, res) => {
+    const translationId = req.params.id;
+    const userId = req.user.id;
+
+    translationDb.getUserIdByTranslationId(translationId).then(userIdTrans => {
```

Comment:
To improve this code in your delete route, try to not exceed 2 levels as a general rule, if you exceed, then extract in a different function. Also try to not have nested promises, as it’s difficult to read, try to have all your then() in the same level.